Chargebee/Classes/Purchase/CBPurchaseManager.swift

```diff
@@ -19,13 +19,13 @@ public class CBPurchase: NSObject {
     var productRequest: SKProductsRequestFactory = SKProductsRequestFactory()
 
     var restoredPurchasesCount = 0
-    private var activeProduct: SKProduct?
+    private var activeProduct: CBProduct?
     var customer: CBCustomer?
     
     var restoreResponseHandler: ((Result<[InAppSubscription], RestoreError>) -> Void)?
     var refreshHandler: RestoreResultCompletion<String>?
     var includeInActiveProducts = false
-    var productType: ProductType?
+    public var productType: ProductType?
```

Comment:
This shouldn't be public.

Reply:
updated to private and added check internally for retry